app/controllers/service_controller.rb

```diff
@@ -150,6 +157,22 @@ def service_form_fields
     }
   end
 
+  def generic_object
+    return unless init_show_variables
+
+    id = params[:show] || params[:x_show]
+    if id.present?
+      @lastaction = "generic_object"
+      @item = @record.generic_objects.find(from_cid(id)).first
+      drop_breadcrumb(:name => _("%{name} (All Generic Objects)") % {:name => @record.name},
+                      :url  => show_link(@record, :display => @display))
+      drop_breadcrumb(:name => @item.name, :url => "/#{controller_name}/show/#{@record.id}?display=generic_objects/show=#{id}")
+      @view = get_db_view(GenericObject)
+      @sb[:rec_id] = id
```

Comment:
why is this needed? so we really need sesssion here?

Reply:
I see it used above in `button` by why is the normal mechanism used in all the buttons used here?

By doing this it seems there's a new place where opening multiple windows would break the app.